lab_4_summarization_textrank/main.py

```diff
@@ -84,31 +95,58 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
         """
         Constructs all the necessary attributes
         """
-        pass
+        super().__init__(stop_words, punctuation)
+        if not (isinstance(stop_words, tuple) and all(isinstance(i, str) for i in stop_words)):
+            raise ValueError
+        self._stop_words = stop_words
+        if not (isinstance(punctuation, tuple) and all(isinstance(j, str) for j in punctuation)):
+            raise ValueError
+        self._punctuation = punctuation
 
     def _split_by_sentence(self, text: str) -> tuple[Sentence, ...]:
         """
         Splits the provided text by sentence
         :param text: the raw text
         :return: a sequence of sentences
         """
-        pass
+        if not isinstance(text, str):
+            raise ValueError
+        final_sentences = []
+        clean_txt = ''
+        new_txt = text.replace('!', '.').replace('?', '.')
+        for idx, elem in enumerate(new_txt[:-2]):
```

Comment:
why `[:-2]`?

Reply:
i made [:-1] now, but i think -2 is also fine, because a word consists at least of 1 letter and the end of the text would be {some letter} and ., or ?, so there's no need to check for further combination of those 3 elements, this combination just won't be there

Reply:
oh i see thanks